### DISCUSSION TITLE
Berechnung Thermische/Elektrische Leistung Vitocal

### DISCUSSION THREAD

**Discussion by Philip-Wiege** [open3e/open3e#161] [General]
Hi @SanMiggel @bomilkar. Hier können wir weiterquatschen zur thermischen Leistung. Die Vitocal hat in der Inneneinheit einen Volumenstromsensor und kennt Vorlauf und Rücklauftemperatur. Lustigerweise weicht die Leistung aber immer etwas von der theoretischen Leistung die man daraus errechnet ab. Ich denke fast, dass die Leistung nicht auf der Wasserseite sondern auf der Kältekreisseite berechnet wird. Die elektrische Seite wird bestimmt kennfeldbasiert geschätzt. Ich gehe stark davon aus, dass es keine echten Stromsensoren gibt. Dafür weicht die angegebene Leistung zu stark von meinen Shellies ab.

**Comment by SanMiggel** [comment]
So genau hab ich mir das noch gar nicht angeschaut... 
Aber genau den Volumenstrom Sensor meinte ich. 

Elektrisch nach Kennfeld.... Hmm. Bei Vereisung passt das aber dann nicht mehr. Ich bilde mir ein dass der instantaneous cop fällt wenn der Verdampfer vereist ist...

**Comment by Philip-Wiege** [comment]
Elektrisch nach Kennfeld hieße dann aber in Abhängigkeit der AC Spannung / interner DC Spannung / Taktverhältnisse der MOSFETs / IGBTs. Da geht schon was und es wird mit recht viel Aufwand vermieden zusätzliche Sensoren einzubauen.

Der instantane COP fällt bei Vereisung tatsächlich ab. Ist ja physikalisch auch sinnvoll, da die Verdampfertemperatur in den Keller geht. In dem Fall muss der Verdichter vom kälteren Verdampfer auf die Soll-Vorlauftemperatur hochpumpen. Dafür muss die Regelung die Verdichterleistung erhöhen, um eine konstante thermische Leistung an den Wasserkreis zu halten. Ergebnis ist dann ein schlechterer COP. Sieht man auf meiner Aufzeichnung ganz gut. Der Verdichter muss immer höher drehen für das halten der Leistung. Dann entscheidet die Steuerung, dass abgetaut werden muss. Der Verdichter dreht volle Kann auf, um den internen Puffer aufzuladen. Dadurch kackt natürlich die Verdampfertemperatur ins bodenlose ab und der Verdampfer vereist total. Dann wird der Kältekreis umgedreht und der Verdampfer wird geheizt. Lustigerweise zeigt die Steuerung eine positive thermische Leistung bei Kältekreisumkehr obwohl dem internen Puffer Wärme entzogen wird. Im Kühlbetrieb bekommt man ebenfalls positive Leistungen obwohl eigentlich dem Heizkreis Wärme entzogen wird.

![image](https://github.com/user-attachments/assets/33e96e80-5805-4d94-8c7b-d84515b63f0b)

**Comment by bomilkar** [comment]
Ich habe mal die Werte zur Leistungsberechnung aufgezeichnet. Bei mir ist die Diskrepanz schon mehr als "ein Wenig".
So habe ich es gemacht: über 24 Stunden habe ich alle 10 Minuten folgende Werte aufgezeichnet:
vcal 1043 AllengraSensor
vcal 268 FlowTemperatureSensor
vcal 269 ReturnTemperatureSensor
vcal 2496 CurrentThermalCapacitySystem

Dann rechne ich:
vcal_1043 * (vcal_268 - vcal_269) * 1.16222
und vergleiche es mit vcal_2496

Das Ergebnis, als Mittelwert über die 24 Stunden:
berechnet: 4567 kW
vcal_2496: 6050 kW

Der Unterschied ist mehr als nur ein paar Prozent. Rechne ich falsch? Oder ist da was faul? Wie sieht das bei euch aus?
Ich habe eine 250A.

**Reply by Philip-Wiege** [reply]
Ich habs bei mir auch mal gerade für ein paar Punkte nachgerechnet. Auch bei mir ist die ausgegebene Leistung deutlich über der selbst berechneten. Sonderpunkte wie Abtauen könnten die Bilanz verfälschen, je nachdem wo die Sensoren sitzen. Das ist aber schon eine merkwürdig große Diskrepanz. Ganz aus dem Bauch heraus würde ich mal vermuten, dass die Volumenstrommessung wohl die komplizierteste ist und eventuell hohe Toleranzen hat. Der muss ja für die Regelung selbst auch nicht präzise ermittelt werden.

**Comment by bomilkar** [comment]
Wenn der Name hinter DID 1043 bedeutet, dass die tatsächlich einen Alengra Ultraschall Durchflusssensor verwenden, dann kann man von wenigen Prozent Ungenauigkeit ausgehen. Keinesfalls erklärt das die 32% Fehler, die ich beobachte.
Was aber durchaus ein Problem sein kann: der Ort, an dem die Temperaturdifferenz (VL - RL) gemessen wird. Wenn für DID 2496 in der Außeneinheit gemessen würde und 268 bzw 269 in der Inneneinheit, dann ist alles möglich: Der Verlust (Temperaturabfall) zwischen Innen und Außen kann beliebig sein. Man müsste also wissen, wo die Temperatursensoren angebracht sind.

**Comment by bomilkar** [comment]
Ich sehe alle paar Samples, dass VL - RL negativ wird. Ziemlich sicher ist das ein Abtauvorgang.
Wenn ich das berücksichtige,  liegt die "offizielle" thermische Leistung immer noch 20% über der berechneten.

**Reply by philippoo66** [reply]
Das wäre nicht das erste Mal, dass Viessmann irgendwas schönrechnet. Wenn man sich die Wirkungs-Koeffizienten anguckt, sind die ja eigentlich auch fast zu schön um wahr zu sein. Als ich die [Grafik zur Visualisierung der Vitocal Energiematrizen zur monatlichen Energiebilanz](https://community.viessmann.de/t5/Konnektivitaet/CAN-Bus-Home-Automation-E3-Generation-lokal-und-kostenlos/m-p/356066/highlight/true#M44941) in den Thread aufnehmen sollte, hatte ich ja auch erstmal gewisse Zweifel zum Ausdruck gebracht,,,

**Reply by Philip-Wiege** [reply]
Ich hab in HomeAssistant mal die berechnete Leistung mitlaufen lassen und gegen die ausgegebene Leistung geplottet:
![image](https://github.com/user-attachments/assets/848a6a31-ed32-4607-8f03-588d244ce30a)

Ich glaube nicht, dass da so viel Leistung "Verloren" geht. Ich bleib dabei, dass wahrscheinlich der Volumenstrom einfach nicht passt. Mich würde es nicht wundern, wenn Viessmann die Leistung aus dem Kältekreis schätzt, was eventuell genauer ist.

**Comment by wangnick** [comment]
Ich glaube, dass Viessmann die Salden der thermischen Energie aus 2496 integriert. Und 2496 scheint mir beim Abtauen positiv zu sein, wird also als eine erbrachte Kühlleistung gewertet, und nicht als eine Reduktion der schon erbrachten Heizleistung! Ich werde das mal genauer unter die Lupe nehmen, auch 2494, und die Werte auch mit 1043*(268-269)*1.16 und denen meines C5-IUF zwischen IDU und ODU vergleichen.

**Comment by bomilkar** [comment]
In den kommenden Tagen wird es zumindest hier warm genug, sodass nicht mehr abgetaut werden sollte. Dann sollte die Auswirkung der Abtauvorgänge gegen 0 gehen. Mal sehen.

Was da als thermische "Energie" angezeigt wird, scheint mir sehr nahe an 2494 zu sein. Aber das liegt deutlich über den realen Werten, die als 1043*(268-269)*1.16 berechnet werden sollten. 268, 269 und 1043 sind auf der Gerätestatusseite zu finden. Obwohl das ViCare schwer zu verfolgen ist. Viguide zeigt 269 NICHT! Vielleicht ist das ja Absicht.

Ich habe heute dies hier gepostet um vielleicht ein wenig Klarstellung (von Viessmann) zu erhalten:
https://community.viessmann.de/t5/Konnektivitaet/ViCare-Energiecockpit-Leistungsberechnung-fehlerhaft/m-p/515022#M63981
Es wäre gut, wenn man einen Viessmann-Vertreter zu einer Aussage bewegen könnte...

**Comment by wangnick** [comment]
Ich habe bei mir mal den Monat Dezember 2024 ausgewertet:
- Laut ViCare komme ich auf eine thermische Energie von 3.07MWh, davon 2.75MWh Heizen und 312kWh Warmwasser. 
- Dies entspricht den beiden Energiesalden 1211.2 (GeneratedCentralHeatingOutputCurrentMonth) 2755kWh und 1391.2 (GeneratedDomesticHotWaterOutputCurrentMonth) 312KWh.
- Die Integration des Leistungswertes 2496 (CurrentThermalCapacitySystem) über die Zeit ergibt bei mir 3067kWh, also exakt der Summe der beiden obigen Salden.
- Die Integration des Produkts aus Volumenstrom 1043.0 (AllengraSensor) und der Differenz zwischen 268.0 (FlowTemperatureSensor) und 269.0 (ReturnTemperatureSensor) ergibt (errechnet aus Kelvinliter * 1.16) 2602kWh
- Der Wärmemengenzähler C5-IUF misst 2768kWh. Er ist allerdings nicht rücklaufend und ignoriert daher Abtauvorgänge.
- Die Integration des Produkts aus den Wärmemengenzählermessungen von Volumenstrom und Differenz zwischen Vor- und Rücklauftemperatur ergibt 2722KWh, inklusive eines Integrals negativer Leistungswerte während des Abtauens von -58kWh. Die gelieferte Wärmeenergie ohne Berücksichtigung des Abtauens beträgt danach also 2780kWh.
- Der Allengrasensor misst knapp 5% Volumenstrom weniger als der Wärmemengenzähler. Dies erklärt die Abweichung der daraus ermittelten Energiemenge zu der des Wärmemengenzählers.

Hier zwei Charts eines Abtauvorgangs von gestern, einmal aus der Vitocal und einmal vom Wärmemengenzähler. Man sieht sehr deutlich, wie die Abtauleistung fälschlich positiv in 2496 dargestellt und in 1211.2 saldiert wird, und wie 2496 auch ansonsten immer mal wieder reichlich von der tatsächlichen Leistung abweicht. VolumeTempdiffDeliveredThermalPower ist genau 1043*(268-269)*1.16:
![vcal250a-abtauleistung-vcal-202501231500](https://github.com/user-attachments/assets/c104a199-dfc4-4927-bc56-f709e5532e65)
![vcal250a-abtauleistung-wmzwp-202501231500](https://github.com/user-attachments/assets/090a32b9-4121-4a47-a39d-b9dc6fa546ee)

Die Aufzeichnung von 2494 hab ich gerade erst eingebaut, da muss ich erst noch auf einen Abtauvorgang warten ...

LG, Sebastian

**Comment by bomilkar** [comment]
Saubere Arbeit!

Ich nehme an, bei der Messung mit dem Wärmemengenzähler hast du die Temperaturdifferenz nahe bei der Inneneinheit gemessen. Hältst du es für denkbar, dass 1211.2+1391.2 in der Außeneinheit gemessen wird? Dass also die Differenz zu 1043*(268-269)*1.16 durch Leitungsverluste zwischen Außen- und Inneneinheit entsteht?

Die Energie zum Abtauen verlässt Haus, muss also vom Ertrag abgezogen werden. Beim Integrieren von 1043*(268-269)*1.16 wird das automatisch berücksichtigt, da (268-269) negativ wird.

Mich beschäftigt die Frage, wie erklärt man die Differenz zwischen Anzeige und Realität. Wenn ich das richtig sehe sind es bei dir -13% . Bei mir ist es fast doppelt so viel! Das wäre zu viel um es zu ignorieren.

**Reply by wangnick** [reply]
> Ich nehme an, bei der Messung mit dem Wärmemengenzähler hast du die Temperaturdifferenz nahe bei der Inneneinheit gemessen.

Ja

> Hältst du es für denkbar, dass 1211.2+1391.2 in der Außeneinheit gemessen wird?

Ich sehe im Anlagenschema (meines ist [dieses](https://static.viessmann.com/resources/vitodesk/schemes/4805622_03/Documents/4805622_2308_03.pdf)) in der Außeneinheit keine Volumenstrommessung, weder im Propankreis noch im Wasserkreis. Die Außeneinheit misst zwar die Wassertemperatur am Wasserausgang des Wärmetauschers (DID 1770, in meinem Schema W16), aber nicht am Wassereingang.

Eine mögliche Erklärung wäre dennoch, dass 2494/2496/1211.2/1391.2 aus der Primärleistung der ODU, also der thermischen Leistung des Propankreises, berechnet wirt. Der Volumenstrom ließe sich womöglich aus der Verdichterleistung schätzen. Und tatsächlich steigt 2496 am Beginn eines Heizzyklus und fällt am Ende ca. 10s vor der Reaktion von 1770, was für eine Messung/Schätzung der Primärleistung spräche:
![vcal250a-heizzyklusende-202501251012](https://github.com/user-attachments/assets/fe057cea-4d28-44e2-937f-91caaf8b2469)

Vielleicht stammt diese Art der Berechnung ja noch von der Vitocal 200 Split-Wärmepumpe. Ich stoße hier allerdings an die Grenzen der zeitlichen Auflösung (alle 10s) meiner Datenerfassung.

Das würde dann allerdings bedeuten, dass so wie der Abtauvorgang auch die Verluste der Vitocal 250-A ODU in die kalte Umgebung sowie die Wärmetauscherverluste zwischen Propan und Wasser in der ODU alle als "gelieferte Leistung" gezählt würden! Ob das so im Sinne der Gesetz- und Normenerfinder ist?

> Mich beschäftigt die Frage, wie erklärt man die Differenz zwischen Anzeige und Realität. Wenn ich das richtig sehe sind es bei dir -13% . Bei mir ist es fast doppelt so viel! Das wäre zu viel um es zu ignorieren.

Wenn unsere Vermutung stimmt, dass die Anzeige auf der Primärkreisleistung beruht, und bei deiner Anlage die Diskrepanz viel größer als bei meiner ist, dann solltest du eventuell die Wärmedämmung der ODU-IDU-Verbindungsleitung mal überprüfen. Bei mir ist diese Doppelrohrleitung ~10m lang und verläuft durch ungeheizte Nutzräume, ist aber relativ gut gedämmt, vor allem im Außenbereich nur ganz kurz, und die Dämmschalen stoßen im Außenbereich dicht an Hauswand und auch (nachdem ich dort nachbessern mußte!) an der Geräteunterseite an. und trotzdem habe ich durch den Nagerschutz (eine Schicht Blitzzement zwischen Außenputz und Rohr, wo Dämmung fehlt) dort eine thermische Schwachstelle:
![FLIR_20250114_040315_031](https://github.com/user-attachments/assets/c4d195f8-20f5-438d-9e24-3b91223c93ea)



LG, Sebastian

**Comment by bomilkar** [comment]
Ihm Rahmen der Messgenauigkeit ist das `d.1043*(d.268-d.269)*1.1622` die Wahrheit. Das ist Physik, die man nicht wegdiskutieren kann. Und die Punkte an denen die Temperaturen d.268 und d.269 gemessen werden definieren die Fläche, die Innen von Außen trennt.
Also ist d.2494 ein Phantasie-Wert, der entweder durch schusseliges Programmieren oder durch Vorsatz entstanden ist. Ich denke, das muss man Viessmann unter die Nase reiben!
Was meint ihr?

**Reply by philippoo66** [reply]
ich meine, da hast du vollkommen Recht mit! Irgendeinen Effekt wird das aber nicht haben. Es wird mit ziemlicher Sicherheit nicht einmal eine Stellungnahme oder Antwort dadrauf geben, das kennen wir ja schon. Wenn man möchte, dass das einen Impact hat, müsste man es vielleicht der Verbraucherzentrale oder so unterbreiten. Im Rahmen der Fainess müsst man vielleicht erst das Eine versuchen, bevor man das Andere macht.

**Comment by bomilkar** [comment]
Der Hinweis auf d.1770 ist gut! Ich werde das mal loggen und mit d.268 vergleichen. Gerade jetzt sehe ich einen Delta von 0,1 K. Das  entspricht einem Verlust von 200 W auf der einen Leitung, also 400 W für hin und zurück. Der Fehler in der einmaligen Messung ist sicher zu groß, um daraus was sinnvolles ableiten zu können. Also loggen und Statistik wird helfen.

Die Verbindung zwischen ODU und IDU ist eine 10 m lange Röhre, wie sie angeblich für Fernwärmenetze verwendet wird. Für diese Röhren muss es ja eine technische Spezifikation geben.

Übrigens: die thermische Leistung ist ja (unter Anderem) ein essentieller Wert zur Beurteilung der Leistungsfähigkeit der Anlage. Wenn es sich herausstellt, dass dies systematisch falsch angegeben wird, dann ist das kein unwesentliches Detail. In meinem Buch würde man so was als Betrug bezeichnen.

**Comment by wangnick** [comment]
d.1770 eilt d.268 voraus. Wenn deine ODU-IDU-Verbindung auch 10m lang ist, und auch aus zwei D32-Rohren besteht, dann enthält eines dieser Rohre ca. 7-8 Liter Wasser. Bei einem Volumenstrom von 1200 l/h, also 20 l/min, braucht es also 20-25 Sekunden bis diese 10m durchflossen sind. Und insofern wird jede Temperaturänderung im Vorlauf bei d.268 immer erst 20-25 Sekunden nach der Registrierung bei d.1770 ankommen.

Man sieht das ganz schön wenn die WW-Bereitung einsetzt. Nach Aktivierung der Sekundärpumpe W03 und Umschalten des IDU-3-Wege-Ventils W17 auf den WW-Vorlauf fällt zunächst d.269 ganz kurz durch abgekühltes Wasser aus dem Rohr zwischen Puffer und IDU bzw. durch kühles HZ-Rücklaufpufferwasser während des Umschaltens des Rücklaufs von HZ auf WW, dann steigt d.269 (und d.1043.1) durch das warme WW-Rücklaufpufferwasser schnell an. Ungefähr 20 Sekunden später hat dieses warme Wasser den Wärmetauscher der ODU Y59 durchflossen und d.1770 beginnt zu steigen. Und weitere 20 Sekunden später erreicht dieses warme Wasser d.268 in der IDU. Und in dieser ganzen Zeit ist der Kompressor noch gar nicht in Betrieb.

![vcal250a-schema-did](https://github.com/user-attachments/assets/8471ee30-3840-4b3d-a48b-c2d798f84def)

![vcal250a-steeprise-tempsensors](https://github.com/user-attachments/assets/ce395eac-8765-4728-87a9-b1033cf352d9)

Wenn allerdings ein stabiles Temperaturniveau herrscht, und d.1770 bei dir 0.1K oberhalb von d.268 liegt, dann muss dass nicht unbedings einen Wärmeverlust auf der Vorlaufleitung bedeuten. Bei mir liegen die Werte von d.1770 in einem stabilen Zustand alle um ca. 0.15K unter denen von d.268. Ich erkläre mir das eher mit dem Grad an Unrichtigkeit der Sensoren.

LG, Sebastian

**Reply by bomilkar** [reply]
Bei mir liegt der Mittelwert der Differenzen zwischen d.1770 und d.268 bei wenigen hundertstel Grad.
Die Messgenauigkeit, insbesondere der Temperaturmessungen, scheint nicht gut genug zu sein, um mit der Differenz zwischen d.1770 und d.268 was anfangen zu können. Vermutlich verbaut Viessmann NTC oder PT Fühler, obwohl das schon längst nicht mehr Stand der Technik ist. Wahrscheinlich trifft das auch auf die Sensoren für d.268 und d.269 zu. Zu dumm!

**Comment by wangnick** [comment]
Es geht ja nicht nur um Stand der Technik. Bei Wärmemengenzählern werden, soweit ich es verstanden habe, die Temperaturmeßfühler jeweils paarweise selektiert.
